rfd/0202-db-multi-session-mfa.md

```diff
@@ -0,0 +1,309 @@
+---
+authors: STeve Huang (xin.huang@goteleport.com)
+state: draft
+---
+
+RFD 202 - Database Multi-session MFA
+
+Required Approvers:
+- Engineering: @r0mant && @codingllama
+
+## What
+
+Allows executing multiple database connections with a single MFA tap.
+
+## Why
+
+Teleport today supports per-session MFA for enhanced security. However, when a
+user needs to run queries on multiple database hosts simultaneously, they have
+to perform a tap for every connection.
+
+A relaxed mode of per-session MFA will be introduced so that an MFA challenge is
+still required for connecting to target databases but the MFA response can be
+reused for a short period of time without the need to prompt the user again in
+that period.
+
+In addition to expanding MFA functionalities, a new `tsh` command will be
+introduced to assist executing queries on multiple database servers using a
+single CLI command.
+
+## Details
+
+### UX
+
+I want to execute a database query on multiple database services without being
+prompted by MFA for every database service. The Teleport role that grants
+database access can be updated as below:
+```diff
+kind: role
+version: v7
+metadata:
+  name: example-role-with-mfa
+spec:
+  options:
+-   require_session_mfa: true
++   # "multi-session" allows reuse of the MFA response where applicable (e.g.
++   # a single WebAuthn prompt for `tsh db exec` on multiple databases), and
++   # fallbacks to per-session MFA in other places where multi-session is not
++   # supported.
++   require_session_mfa: "multi-session"
+  allow:
+    db_labels:
+      'env': 'dev'
+    db_users: ["mysql"]
+```
+
+To execute the same query on multiple databases:
+```bash
+$ tsh db exec "select @@hostname" --db-user mysql --dbs mysql-db1,mysql-db2
+MFA is required to execute database sessions
+Tap any security key
+Detected security key tap
+
+Executing command for 'mysql-db1':
+@@hostname
+mysql-db1-hostname
+
+Executing command for 'mysql-db2':
+@@hostname
+mysql-db2-hostname
```

Comment:
Does this imply this will execute sequentially, unless `--max-connections` flag is provided?